src/main/java/com/github/sdonkov/PrefixTree.java

```diff
@@ -75,7 +80,115 @@ public boolean contains(CharSequence word) {
         return false;
     }
 
-    public boolean add(CharSequence word) {
+    @Override
+    public int size() {
+        return countNodes(root);
+    }
+
+    private int countNodes(Node node) {
+        int count = node.isEndOfWord() ? 1 : 0;
+        for (Node child : node.edges) {
+            count += countNodes(child);
+        }
+        return count;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size() == 0;
```

Comment:
It'd be way faster (in case it is not empty) to check whether the root node has any descendents and is not an end-of-word itself. This would work only if we get rid of the removal or alter it so that it removes edges which are not end-of-words.